lighthouse-core/audits/byte-efficiency/render-blocking-resources.js

```diff
@@ -135,38 +148,43 @@ class RenderBlockingResources extends Audit {
     const originalEstimate = simulator.simulate(fcpGraph).timeInMs;
 
     let totalChildNetworkBytes = 0;
-    const minimalFCPGraph = fcpGraph.cloneWithRelationships(node => {
+    const minimalFCPGraph = /** @type {NetworkNode} */ (fcpGraph.cloneWithRelationships(node => {
+      // If a node can be deferred, exclude it from the new FCP graph
       const canDeferRequest = deferredIds.has(node.id);
+      if (node.type !== Node.TYPES.NETWORK) return !canDeferRequest;
+
+      const networkNode = /** @type {NetworkNode} */ (node);
+
       const isStylesheet =
-        node.type === Node.TYPES.NETWORK &&
-        node.record._resourceType === WebInspector.resourceTypes.Stylesheet;
+        networkNode.record._resourceType === WebInspector.resourceTypes.Stylesheet;
       if (canDeferRequest && isStylesheet) {
         // We'll inline the used bytes of the stylesheet and assume the rest can be deferred
-        const wastedBytes = wastedCssBytesByUrl.get(node.record.url) || 0;
-        totalChildNetworkBytes += node.record._transferSize - wastedBytes;
+        const wastedBytes = wastedCssBytesByUrl.get(networkNode.record.url) || 0;
+        totalChildNetworkBytes += (networkNode.record._transferSize || 0) - wastedBytes;
       }
-
-      // If a node can be deferred, exclude it from the new FCP graph
       return !canDeferRequest;
-    });
+    }));
 
     // Add the inlined bytes to the HTML response
-    minimalFCPGraph.record._transferSize += totalChildNetworkBytes;
+    const originalTransferSize = minimalFCPGraph.record._transferSize;
+    const safeTransferSize = originalTransferSize || 0;
+    minimalFCPGraph.record._transferSize = safeTransferSize + totalChildNetworkBytes;
     const estimateAfterInline = simulator.simulate(minimalFCPGraph).timeInMs;
-    minimalFCPGraph.record._transferSize -= totalChildNetworkBytes;
+    minimalFCPGraph.record._transferSize = originalTransferSize;
     return Math.round(Math.max(originalEstimate - estimateAfterInline, 0));
   }
 
   /**
-   * @param {!Artifacts} artifacts
+   * @param {LH.Artifacts} artifacts
    * @param {LH.Audit.Context} context
-   * @return {Map<string, number>}
+   * @return {Promise<Map<string, number>>}
    */
   static async computeWastedCSSBytes(artifacts, context) {
     const wastedBytesByUrl = new Map();
     try {
       // TODO(phulce): pull this out into computed artifact
       const results = await UnusedCSS.audit(artifacts, context);
+      // @ts-ignore - TODO(bckenny): details types.
```

Comment:
we *could* have a more specific type returned from `UnusedCSS.audit` to avoid the `any` type, but it unfortunately passes through `audit` in the super class, making the return type generic as well. We could figure this out with details types or the above TODO about moving to a computed artifact would fix it as well.

Reply:
yeah let's just do the right computed artifact fix later 👍